javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/config/ConfigurationFileLoader.java

```diff
@@ -76,13 +77,71 @@ static Map<String, String> loadConfigFile() {
           configurationFilePath);
     }
 
+    SubstitutionReplacer substitutorReplacer = new SubstitutionReplacer(configurationFilePath);
+
     return properties.entrySet().stream()
-        .collect(Collectors.toMap(e -> e.getKey().toString(), e -> e.getValue().toString()));
+        .collect(
+            Collectors.toMap(
+                e -> e.getKey().toString(),
+                e -> substitutorReplacer.replace(e.getValue().toString())));
   }
 
   @Override
   public int order() {
     // make sure it runs after all the user-provided customizers
     return Integer.MAX_VALUE;
   }
+
+  static class SubstitutionReplacer {
```

Comment:
@mateuszrzeszutek please take a look at this implementation.
As implemented, it will replace *any* occurrence of `$$` with `$`, _even when not followed by `{placeholder}`_.  This could well be a problem.  It is not impossible that someone could have `$$` somewhere in a property value.
I guess we could say only escape `$`s when immediately followed by a substitution match???